src/libraries/System.Runtime/tests/System/Attributes.cs

```diff
@@ -533,6 +533,35 @@ public static void NegTest8()
             AssertExtensions.Throws<ArgumentException>(null, () => (ArgumentUsageAttribute)Attribute.GetCustomAttribute(paramInfos[0], attributeType, false));
 
         }
+
+        // reproduces https://github.com/dotnet/runtime/issues/64335
+        [Fact]
+        [SkipOnMono("Mono throws on open generic type passed to GetCustomAttributes")]
+        public static void GetCustomAttributesDoesNotReturnNullForOpenGenericType()
+        {
+            Attribute[] openGenericAttributes = Attribute.GetCustomAttributes(typeof(HasGenericAttribute), typeof(GenericAttribute<>));
```

Comment:
Please add test/fix for all member types including assembly, i got `InvalidCastException` when trying to get generic attribute from assembly
```log
System.InvalidCastException
  HResult=0x80004002
  Message=Unable to cast object of type 'System.Object[]' to type 'System.Attribute[]'.
  Source=System.Private.CoreLib
  StackTrace:
   at System.Runtime.CompilerServices.CastHelpers.ChkCastAny(Void* toTypeHnd, Object obj)
   at System.Attribute.GetCustomAttributes(Assembly element, Type attributeType, Boolean inherit)
   at System.Attribute.GetCustomAttributes(Assembly element, Type attributeType)
   at ConsoleApp13.Program.Main(String[] args) in \ConsoleApp13\Program.cs:line 22
```